go/vt/vttablet/tabletserver/txthrottler/tx_throttler.go

```diff
@@ -224,6 +226,20 @@ func (t *txThrottler) Open() (err error) {
 		return nil
 	}
 	log.Info("txThrottler: opening")
+
+	// get cells from topo if none defined in tabletenv config
+	if len(t.config.healthCheckCells) == 0 {
+		ctx, cancel := context.WithTimeout(context.Background(), topo.RemoteOperationTimeout)
+		defer cancel()
+
+		if t.config.healthCheckCells, err = t.topoServer.GetKnownCells(ctx); err != nil {
+			return err
+		} else if len(t.config.healthCheckCells) == 0 {
```

Comment:
isn't this condition `if len(t.config.healthCheckCells) == 0` nested within an identical `if len(t.config.healthCheckCells) == 0` a few lines above?

Reply:
@shlomi-noach I'll see if I can make this more readable. The intent here is:
1. If no cells are defined we get all known cells from the topology (first `if len(...)` condition)
2. After we get known cells from the topology we check if we got zero cells again (second `if len(...)` condition) because this would create a throttler that never finds tablets

> what' the final word about re-evaluating all cells dynamically?

I think dynamic is still the goal but it would require larger changes that I haven't been able to wrap my head around. Currently the cells to watch is passed to `discovery.NewHealthCheck(...)` once and this object is passed around to each topology watcher

I think we'd need to re-init the `discovery.HealthCheck` and re-pass the new object to each topo watcher. This may may mean restarting each topo watcher from scratch, etc. I was hoping to defer this to later as dynamic cells is a nice to have for our use case but it could be tackled here

Reply:
Oh, I missed the fact that `t.config.healthCheckCells` is being re-read. Strike my above comment.

Reply:
It seems like `.GetKnownCells(...)` might not check if it's returning an empty slice, so I think the 2nd length-check should stay for now 👍

On the dynamic cells, any thoughts on that approach is appreciated. I can tackle that in this PR or another 👍

Reply:
@shlomi-noach thanks, that makes sense

In this throttler I think we'll need to tear down every topology watcher (that are passed the single `discovery.HealthCheck`) created by `healthCheckFactory`, recreate the `discovery.HealthCheck` with a new list of `cellsToWatch` and then recreate a topology watcher for every cell when cells change. I think that will need some mutexes that I can hopefully avoid in the actual `.Throttle(...)` path 👍 

I think that's all do-able but I wonder how to react if any of that logic fails in the background. We can log that it happened but nothing will be waiting for an error 🤔